src/RA_md5factory.cpp

```diff
@@ -1,66 +1,69 @@
 
-    memset(buffer, 0, MD5_STRING_LEN + 1);
-    sprintf_s(buffer, MD5_STRING_LEN + 1,
-        "%02x%02x%02x%02x%02x%02x%02x%02x"
-        "%02x%02x%02x%02x%02x%02x%02x%02x",
-        digest[0], digest[1], digest[2], digest[3], digest[4], digest[5], digest[6], digest[7],
-        digest[8], digest[9], digest[10], digest[11], digest[12], digest[13], digest[14], digest[15]);
+    md5_init(pms.get());
+    md5_append(pms.get(), pDataBuffer.get(), sStringToMD5.length());
+    md5_finish(pms.get(), digest.get());
 
-    delete[] pDataBuffer;
-    pDataBuffer = nullptr;
+    std::memset(static_cast<void*>(buffer), 0, ra::MD5_STRING_LEN + 1U);
+    sprintf_s(buffer, ra::MD5_STRING_LEN + 1U,
+              "%02x%02x%02x%02x%02x%02x%02x%02x"
+              "%02x%02x%02x%02x%02x%02x%02x%02x",
+              digest[0], digest[1], digest[2], digest[3], digest[4], digest[5], digest[6], digest[7],
+              digest[8], digest[9], digest[10], digest[11], digest[12], digest[13], digest[14], digest[15]);
 
     return buffer;
 }
 
-std::string RAGenerateMD5(const BYTE* pRawData, size_t nDataLen)
+_Use_decl_annotations_
+std::string RAGenerateMD5(const BYTE* pRawData, size_t nDataLen) noexcept
 {
-    static char buffer[33];
-
-    md5_state_t pms;
-    md5_byte_t digest[16];
-
-    static_assert(sizeof(md5_byte_t) == sizeof(BYTE), "Must be equivalent for the MD5 to work!");
-
-    md5_init(&pms);
-    md5_append(&pms, static_cast<const md5_byte_t*>(pRawData), nDataLen);
-    md5_finish(&pms, digest);
-
-    memset(buffer, 0, MD5_STRING_LEN + 1);
-    sprintf_s(buffer, MD5_STRING_LEN + 1,
-        "%02x%02x%02x%02x%02x%02x%02x%02x"
-        "%02x%02x%02x%02x%02x%02x%02x%02x",
-        digest[0], digest[1], digest[2], digest[3], digest[4], digest[5], digest[6], digest[7],
-        digest[8], digest[9], digest[10], digest[11], digest[12], digest[13], digest[14], digest[15]);
-
-    return buffer;	//	Implicit promotion to std::string
+    char buffer[33]{};
+    auto pms{ std::make_unique<md5_state_t>() };
+    auto digest{ std::make_unique<md5_byte_t[]>(16) };
+   
+    static_assert(ra::is_same_size_v<md5_byte_t, BYTE>, "Must be equivalent for the MD5 to work!");
+
+    md5_init(pms.get());
+    md5_append(pms.get(), static_cast<const md5_byte_t*>(pRawData), ra::to_signed(nDataLen));
+    md5_finish(pms.get(), digest.get());
+
+    std::memset(static_cast<void*>(buffer), 0, ra::MD5_STRING_LEN + 1U);
+    sprintf_s(buffer, ra::MD5_STRING_LEN + 1U,
+              "%02x%02x%02x%02x%02x%02x%02x%02x"
+              "%02x%02x%02x%02x%02x%02x%02x%02x",
+              digest[0], digest[1], digest[2], digest[3], digest[4], digest[5], digest[6], digest[7],
+              digest[8], digest[9], digest[10], digest[11], digest[12], digest[13], digest[14], digest[15]);
+
+    return buffer; // Implicit promotion to std::string
 }
 
-std::string RAGenerateMD5(const std::vector<BYTE> DataIn)
+std::string RAGenerateMD5(const std::vector<BYTE>& DataIn) noexcept
```

Comment:
this was being passed by value, very slow for structures with copy semantics. Wasn't sure if was desired to just move it into the function so it's still const.